src/coreclr/pal/src/thread/thread.cpp

```diff
@@ -1474,138 +1468,51 @@ SetThreadDescription(
-        pThread,
-        hThread,
-        lpThreadDescription
-        );
-
-    if (NO_ERROR != palError)
-    {
-        pThread->SetLastError(palError);
-    }
-
-    LOGEXIT("SetThreadDescription");
-    PERF_EXIT(SetThreadDescription);
-
-    return HRESULT_FROM_WIN32(palError);
-}
+    CPalThread *pThread = InternalGetCurrentThread();
 
-PAL_ERROR
-CorUnix::InternalSetThreadDescription(
-    CPalThread *pThread,
-    HANDLE hTargetThread,
-    PCWSTR lpThreadDescription
-)
-{
-    PAL_ERROR palError = NO_ERROR;
     CPalThread *pTargetThread = NULL;
     IPalObject *pobjThread = NULL;
-    int error = 0;
-    int maxNameSize = 0;
     int nameSize;
     char *nameBuf = NULL;
 
-// The exact API of pthread_setname_np varies very wildly depending on OS.
-// For now, only Linux, macOS and FreeBSD are implemented.
-#if defined(__linux__) || defined(__APPLE__) || defined(__FreeBSD__)
-
-    palError = InternalGetThreadDataFromHandle(
-        pThread,
-        hTargetThread,
-        &pTargetThread,
-        &pobjThread
-        );
-
-    if (NO_ERROR != palError)
-    {
-        goto InternalSetThreadDescriptionExit;
-    }
-
-    pTargetThread->Lock(pThread);
-
-    // Ignore requests to set the main thread name because
-    // it causes the value returned by Process.ProcessName to change.
-    if ((pid_t)pTargetThread->GetThreadId() == getpid())
-    {
-        goto InternalSetThreadDescriptionExit;
-    }
-
-    /* translate the wide char lpThreadDescription string to multibyte string */
-    nameSize = WideCharToMultiByte(CP_ACP, 0, lpThreadDescription, -1, NULL, 0, NULL, NULL);
-
-    if (0 == nameSize)
-    {
-        palError = ERROR_INTERNAL_ERROR;
-        goto InternalSetThreadDescriptionExit;
-    }
-
-    nameBuf = (char *)malloc(nameSize);
-    if (nameBuf == NULL)
-    {
-        palError = ERROR_OUTOFMEMORY;
-        goto InternalSetThreadDescriptionExit;
-    }
-
-    if (WideCharToMultiByte(CP_ACP, 0, lpThreadDescription, -1, nameBuf, nameSize, NULL,
-                            NULL) != nameSize)
-    {
-        palError = ERROR_INTERNAL_ERROR;
-        goto InternalSetThreadDescriptionExit;
-    }
-
-    // Null terminate early.
-    // pthread_setname_np only accepts up to 16 chars on Linux,
-    // 64 chars on macOS and 20 chars on FreeBSD.
-    if (nameSize > MAX_THREAD_NAME_SIZE)
-    {
-        nameBuf[MAX_THREAD_NAME_SIZE] = '\0';
-    }
-
-    #if defined(__linux__) || defined(__FreeBSD__)
-    error = pthread_setname_np(pTargetThread->GetPThreadSelf(), nameBuf);
-    #endif
-
-    #if defined(__APPLE__)
-    // on macOS, pthread_setname_np only works for the calling thread.
-    if (PlatformGetCurrentThreadId() == pTargetThread->GetThreadId())
-    {
-        error = pthread_setname_np(nameBuf);
-    }
-    #endif
-
-    if (error != 0)
+    PAL_ERROR palError = InternalGetThreadDataFromHandle(pThread, hThread, &pTargetThread, &pobjThread);
+    if (palError != NO_ERROR)
```

Comment:
@am11 I have just discovered that you have reversed this condition by accident. So we effectively never set the name. And if the InternalGetThreadDataFromHandle fails, the code crashes because the pTargetThread is null (I've just investigated such crash and found this to be the culprit)

Reply:
Ops, this was an accident. Thanks for catching this. I will make a fix.